x/profiles/keeper/keeper_blocks.go

```diff
@@ -108,3 +108,17 @@ func (k Keeper) DeleteAllUserBlocks(ctx sdk.Context, user string) {
 		store.Delete(types.UserBlockStoreKey(block.Blocker, block.Subspace, block.Blocked))
 	}
 }
+
+// DeleteSubspaceUserBlocks deletes the user blocks created by the specified user from the subspace having the given id
+func (k Keeper) DeleteSubspaceUserBlocks(ctx sdk.Context, subspaceID, user string) {
```

Comment:
I think that this method can be simply written like this:
```go
// DeleteSubspaceUserBlocks deletes the user blocks created by the specified user from the subspace having the given id
func (k Keeper) DeleteSubspaceUserBlocks(ctx sdk.Context, subspaceID, user string) {
        store := ctx.KVStore(k.storeKey)

	k.IterateSubspaceUserBlocks(ctx, user, subspaceID, func(index int64, block types.UserBlock) (stop bool) {
		store.Delete(types.UserBlockStoreKey(block.Blocker, block.Subspace, block.Blocked))
		return false
	})
}
```
Also we should add some unit tests for it